uwrt_mars_rover_drivetrain/uwrt_mars_rover_drivetrain_hw/launch/drivetrain.launch.py

```diff
@@ -0,0 +1,73 @@
+from ament_index_python.packages import get_package_prefix, get_package_share_path
+from launch import LaunchDescription
+from launch.actions import RegisterEventHandler
+from launch.event_handlers import OnProcessExit
+from launch.substitutions import Command, FindExecutable, LaunchConfiguration, PathJoinSubstitution
+from launch_ros.actions import Node
+from launch_ros.parameter_descriptions import ParameterValue
+
+
+def generate_launch_description():
+    drivetrain_description_package_path = get_package_share_path('uwrt_mars_rover_drivetrain_description')
+    model_path = drivetrain_description_package_path / 'urdf' / 'drivetrain.urdf.xacro'
+    rviz_config_path = drivetrain_description_package_path / 'rviz' / 'urdf.rviz'
+    controllers_config_path = get_package_share_path(
+        'uwrt_mars_rover_drivetrain_hw') / 'config' / 'drivetrain_controllers.yaml'
+
+    robot_description_content = ParameterValue(Command([f'ros2 run xacro xacro ', str(model_path)]), value_type=str)
+    robot_description = {"robot_description": robot_description_content}
+
+    # Nodes
+    nodes = []
+    nodes += [Node(
+        package="controller_manager",
+        executable="ros2_control_node",
+        parameters=[robot_description, controllers_config_path],
+        output={
+            "stdout": "screen",
+            "stderr": "screen",
+        },
+    )]  # TODO: use custom control node w/ RT scheduling(port from ros1 uwrt_mars_rover branch)
```

Comment:
Create issue for this

Reply:
custom controller manager .... with real time capabilities potentially?

Reply:
no not a custom controller manager. just instead of using the node provided by `ros2_control`, we use our own node that is almost the same, except makes the appropriate syscalls to get rt performance on a linux machine with the prempt rt patch. see this:
https://github.com/uwrobotics/uwrt_mars_rover/blob/ros1/uwrt_mars_rover_control/src/uwrt_mars_rover_control_node.cpp

Reply:
#178